map_model/src/map.rs

```diff
@@ -735,6 +740,42 @@ impl Map {
         bail!("Can't find {}", id)
     }
 
+    fn populate_intersection_quad_tree(&self) -> Result<()> {
+        let quad_tree_lock = Arc::clone(&self.intersection_quad_tree);
+        let mut quad_tree = quad_tree_lock.write().unwrap();
+
+        if quad_tree.is_some() {
+            return Ok(());
+        }
+
+        let mut quad: FindClosest<IntersectionID> = FindClosest::new();
+
+        for intersection in self.all_intersections() {
+            quad.add_polygon(intersection.id, &intersection.polygon);
+        }
+        *quad_tree = Some(quad);
+        Ok(())
+    }
+
+    pub fn localise_lon_lat_to_map(&self, point: &LonLat) -> Pt2D {
+        point.to_pt(&self.gps_bounds)
+    }
+
+    pub fn find_i_by_pt2d(&self, pnt: Pt2D) -> Result<IntersectionID> {
+        self.populate_intersection_quad_tree().unwrap();
+        let quad_tree_lock = Arc::clone(&self.intersection_quad_tree);
+        let quad_tree = quad_tree_lock.read().unwrap();
+
+        if let Some(tree) = &*quad_tree {
+            let (intersection_id, _) = tree
+                .closest_pt(pnt, Distance::meters(100.0))
+                .context("Failed to find intersection within 100m of specified point")?;
+            Ok(intersection_id)
+        } else {
+            bail!("Intersection Quad tree was somehow blank even after generation")
```

Comment:
This is fine by me. I'd even be OK using `unreachable` here, or a further `unwrap` above

Reply:
Changed to unreadchable